app/src/main/java/io/apicurio/registry/storage/metrics/StorageMetricsStore.java

```diff
@@ -75,40 +83,78 @@ private static class ArtifactVersionKey {
     }
 
     public void onStart(@Observes StartupEvent ev) {
-        totalSchemasCounters = new CheckPeriodCache<>(Duration.ofMillis(limitsCheckPeriod));
-        artifactsCounters = new CheckPeriodCache<>(Duration.ofMillis(limitsCheckPeriod));
-        artifactVersionsCounters = new CheckPeriodCache<>(Duration.ofMillis(limitsCheckPeriod));
+        createTotalSchemasCache();
+        createTotalArtifactsCache();
+        createTotalArtifactVersionsCache();
+    }
+
+    private void createTotalSchemasCache() {
+        totalSchemaCountersLoader = new CacheLoader<>() {
+            @Override
+            public AtomicLong load(@NotNull String tenantId) {
+                log.info("Initializing total schemas counter, tid {}", tenantContext.tenantId());
+                long count = storage.countTotalArtifactVersions();
+                return new AtomicLong(count); }
+        };
+
+        totalSchemasCounters = CacheBuilder
+                .newBuilder()
+                .expireAfterWrite(limitsCheckPeriod, TimeUnit.MILLISECONDS)
+                .maximumSize(1000)
+                .build(totalSchemaCountersLoader);
+    }
+
+    private void createTotalArtifactsCache() {
+        artifactsCountersLoader = new CacheLoader<>() {
+            @Override
+            public AtomicLong load(@NotNull String tenantId) {
+                log.info("Initializing total schemas counter, tid {}", tenantContext.tenantId());
+                long count = storage.countArtifacts();
+                return new AtomicLong(count); }
+        };
+
+        artifactsCounters = CacheBuilder
+                .newBuilder()
+                .expireAfterWrite(limitsCheckPeriod, TimeUnit.MILLISECONDS)
+                .maximumSize(1000)
+                .build(artifactsCountersLoader);
+    }
+
+    private void createTotalArtifactVersionsCache() {
+        artifactVersionsCountersLoader = new CacheLoader<>() {
+            @Override
+            public AtomicLong load(@NotNull ArtifactVersionKey artifactVersionKey) {
+                long count = storage.countArtifactVersions(artifactVersionKey.groupId, artifactVersionKey.artifactId);
```

Comment:
This (and loaders above) could throw a checked exception. I see we did not handle it before, but *maybe* it's worth re-throwing it as in `TenantContextLoader` and use `get` instead of `getUnchecked`.

Reply:
I'm handling the exception in the TenantContextLoader since the cache loader there can legitimately throw an exception (e.g. a particular tenant is not found) therefore it must be handled, in this case, the exception should not be handled since the cache loader for these caches should not throw any exception (if it happens, it will result in a 500, as it should and reported as a liveness problem).